src/components/Provider.js

```diff
@@ -20,7 +20,7 @@ function warnAboutReceivingStore() {
 
 export default class Provider extends Component {
   getChildContext() {
-    return { store: this.store }
+    return { store: this.store, react15CompatibilityMode: this.props.react15CompatibilityMode }
```

Comment:
I interpreted @gaearon's messages as we wanted to impose the least amount of breaking changes as possible, that's why I made it a global default setting which is "work like it used to, even if that means being not as fast as possible and maybe get stale props". It's certainly less code in react-redux to not have a global setting and require devs to go enable the compat setting on all their connected textboxes. I'm fine with either approach but I just need to be told which way we're going.